frontend/packages/noobaa-storage-plugin/src/components/data-resiliency-card/data-resiliency-card.tsx

```diff
@@ -0,0 +1,130 @@
+      </div>
+    </>
+  ) : (
+    <>
+      <div className="nb-data-resiliency__icon--error">
+        <Icon type="fa" name="exclamation-triangle" size="5x" />
+      </div>
+      <div className="nb-data-resiliency__status-title--error">No data available</div>
+    </>
+  );
+
+const DataResiliencyBuildBody: React.FC<DataResiliencyBuildBody> = ({
+  progressPercentage,
+  eta,
+}) => {
+  const formattedEta = getFormattedEta(eta);
+  return (
+    <>
+      <div className="nb-data-resiliency__title">Rebuilding data</div>
+      <Progress
+        className="nb-data-resiliency__utilization-bar"
+        value={progressPercentage}
+        title="Rebuilding in Progress"
+        label={`${progressPercentage}%`}
+      />
+      <span className="text-secondary nb-data-resiliency__eta">
+        Estimating {formattedEta} to completion
+      </span>
+    </>
+  );
+};
+
+const DataResiliency: React.FC<DashboardItemProps> = ({
+  watchPrometheus,
+  stopWatchPrometheusQuery,
+  prometheusResults,
+}) => {
+  React.useEffect(() => {
+    Object.keys(DATA_RESILIENCE_QUERIES).forEach((key) =>
+      watchPrometheus(DATA_RESILIENCE_QUERIES[key]),
+    );
+    return () =>
+      Object.keys(DATA_RESILIENCE_QUERIES).forEach((key) =>
+        stopWatchPrometheusQuery(DATA_RESILIENCE_QUERIES[key]),
+      );
+  }, [watchPrometheus, stopWatchPrometheusQuery]);
+
+  const rebuildProgressQueryResult = prometheusResults.getIn([
+    DATA_RESILIENCE_QUERIES.REBUILD_PROGRESS_QUERY,
+    'result',
+  ]);
+  const etaQueryResult = prometheusResults.getIn([
+    DATA_RESILIENCE_QUERIES.REBUILD_TIME_QUERY,
+    'result',
+  ]);
+
+  const rebuildProgress = getPropsData(rebuildProgressQueryResult);
+  const eta = getPropsData(etaQueryResult);
+
+  const progressPercentage = rebuildProgress ? Number(rebuildProgress) : null;
+
+  return (
+    <DashboardCard>
+      <DashboardCardHeader>
+        <DashboardCardTitle>Data Resiliency</DashboardCardTitle>
+        <DashboardCardHelp>
+          Data resiliency is the ability of stored objects to recover and continue operating in the
+          case of a failure.Certain changes in the system (unavailable resource/ change of bucket
+          policy etc.) cause an object to require a rebuilding process in order to stay resilient.
+        </DashboardCardHelp>
+      </DashboardCardHeader>
+      <DashboardCardBody
+        className="nb-data-resiliency__dashboard-body"
+        isLoading={!rebuildProgressQueryResult}
+      >
+        {progressPercentage >= 100 || !progressPercentage ? (
+          <DataResiliencyStatusBody isResilient={progressPercentage} />
+        ) : (
+          <DataResiliencyBuildBody progressPercentage={progressPercentage} eta={eta} />
+        )}
+      </DashboardCardBody>
+    </DashboardCard>
+  );
+};
+
+export default withDashboardResources(DataResiliency);
+
+type DataResiliencyBuildBody = {
+  progressPercentage: number;
+  eta: string;
```

Comment:
eta should be an optional property

Reply:
why @cloudbehl . Its coming from metrics and needs to be implemented in UI as per UXD
I guess you are talking about the case wen the eta value dont come from prometheus. I will fix that and dont show eta in that case.
Thoughts ?

Reply:
Correct make sense, Looks good to me. @rawagner your thoughts on this?

Reply:
sounds good to me